File: codeforces/1800~1899/1859/b.cc
#include <bits/stdc++.h>

using namespace std;

using i64 = long long;

void solve() {
  int n = 0;
  cin >> n;
  vector<vector<i64>> a;
  for (int i = 0; i < n; ++i) {
    int m = 0;
    cin >> m;
    vector<i64> row;
    for (int i = 0; i < m; ++i) {
      i64 x = 0;
      cin >> x;
      row.push_back(x);
    }

    sort(row.begin(), row.end());
    a.push_back(row);
  }

  sort(a.begin(), a.end(), [&](auto& r1, auto& r2) {
    return r1[0] < r2[0];
  });

  // move all mins to first array
  i64 first = 0;
  for (int i = 0; i < n; ++i) {
    first += a[i][0];
  }

  for (int i = 1; i < n; ++i) {
    first += a[i][1] - a[i][0];
  }
  i64 ans = first;

  // move all mins to jth array
  i64 second = 0;
  for (int i = 0; i < n; ++i) {
    second += a[i][1];
  }
  for (int j = 1; j < n; ++j) {
    i64 new_ans = second - a[j][1] + a[0][0];
    ans = max(ans, new_ans);
  }

  cout << ans << endl;
}

int main(void) {
  ios::sync_with_stdio(false);
  cin.tie(0);

  int T = 0;
  cin >> T;
  for (int t = 0; t < T; ++t) {
    solve();
  }
  return 0;
}

File: codeforces/1800~1899/1859/b.rs
use std::io;
use std::cmp::max;

fn read_int() -> i32 {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    return s.trim().parse::<i32>().expect("invalid input");
}

fn read_vec() -> Vec<i64> {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    s.split(" ").map(|s| s.trim().parse::<i64>().expect("invalid input")).collect()
}

fn solve() {
    let n = read_int();
    let mut a: Vec<Vec<i64>> = Vec::new();
    for _i in 0..n {
        let _m = read_int();
        let mut row = read_vec();
        row.sort();
        a.push(row);
    }

    a.sort_by_key(|row| row[0]);

    let mut first: i64 = 0;
    for i in 0..n {
        if i == 0 {
            first += a[i as usize][0];
        } else {
            first += a[i as usize][1];
        }
    }

    let mut ans: i64 = first;

    let mut second: i64 = 0;
    for i in 0..n {
        second += a[i as usize][1];
    }
    for i in 1..n {
        let new_ans: i64 = second - a[i as usize][1] + a[0][0];
        ans = max(ans, new_ans);
    }

    println!("{}", ans);
}

fn main() {
    let t = read_int();
    for _t in 0..t {
        solve();
    }
}
